### PR TITLE
fix: use `tailwind.config.mjs` file by default config

### PR DESCRIPTION
**Pull request by torn4dom4n**
#### What kind of changes does this PR include?

- Changes Starlight docs site content

#### Description

Since Astro v3.2.4, the `@astrojs/tailwind` integration now creates a `tailwind.config.mjs` file by default with PR https://github.com/withastro/astro/pull/8638. So this content should be changed.

> I'm participating in Hacktoberfest

### CODE REVIEW

**Review by HiDeoo — COMMENTED** [review]
Thanks for your contribution :raised_hands: 

Some potential thoughts/questions I had while checking out the PR:

- I guess we would want to update the theme editor [generated code](https://github.com/withastro/starlight/blob/main/docs/src/components/theme-designer.astro#L107-L122) too.
- I assume the [Tailwind plugin JSDoc example](https://github.com/withastro/starlight/blob/main/packages/tailwind/index.ts#L12-L31) would also need to be updated?
- Should the [Starlight Tailwind Starter Kit](https://github.com/withastro/starlight/tree/main/examples/tailwind) (config file + README) also be updated?

Also, docs-only changes do not require a changeset, and even if we end up updating the Starlight Tailwind Starter Kit, I'm not sure if that would require a changeset either.

**Comment by delucis** [comment]
Thanks for picking this up @torn4dom4n!

I think all of @HiDeoo’s suggestions here are good ones, and I can confirm we don’t need a changeset for these changes. If we update the JSDoc in the Tailwind plugin though, then we would want a patch changeset for the Tailwind package so that update gets released.

File: docs/src/content/docs/guides/css-and-tailwind.mdx [hunk 1]
```diff
@@ -181,14 +181,14 @@ If you already have a Starlight site and want to add Tailwind CSS, follow these
     });
     ```
 
-5.  Add the Starlight Tailwind plugin to `tailwind.config.cjs`:
+5.  Add the Starlight Tailwind plugin to `tailwind.config.mjs`:
 
     ```js ins={2,7}
-    // tailwind.config.cjs
+    // tailwind.config.mjs
     const starlightPlugin = require('@astrojs/starlight-tailwind');
```
Comment (HiDeoo) [inline docs/src/content/docs/guides/css-and-tailwind.mdx:188]:
```suggestion
    import starlightPlugin from '@astrojs/starlight-tailwind';
```

File: docs/src/content/docs/guides/css-and-tailwind.mdx [hunk 2]
```diff
@@ -206,12 +206,12 @@ If set, the following options will override Starlight’s default styles:
 - `fontFamily.mono` — used for code examples
 
 ```js {12,14,18,20}
-// tailwind.config.cjs
+// tailwind.config.mjs
 const starlightPlugin = require('@astrojs/starlight-tailwind');
 const colors = require('tailwindcss/colors');
```
Comment (HiDeoo) [inline docs/src/content/docs/guides/css-and-tailwind.mdx:211]:
```suggestion
import starlightPlugin from '@astrojs/starlight-tailwind';
import colors from 'tailwindcss/colors';
```

File: examples/tailwind/tailwind.config.mjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const colors = require('tailwindcss/colors');
+const starlightPlugin = require('@astrojs/starlight-tailwind');
```
Comment (HiDeoo) [inline examples/tailwind/tailwind.config.mjs:2]:
```suggestion
import colors from 'tailwindcss/colors';
import starlightPlugin from '@astrojs/starlight-tailwind';
```

File: packages/tailwind/index.ts [hunk 4]
```diff
@@ -9,11 +9,11 @@ import plugin from 'tailwindcss/plugin';
  * - Links Starlight’s fonts to `sans` and `mono` in Tailwind theme settings.
  *
  * @example
- * // tailwind.config.cjs
+ * // tailwind.config.mjs
  * const colors = require('tailwindcss/colors');
  * const starlightPlugin = require('@astrojs/starlight/tailwind');
```
Comment (HiDeoo) [inline packages/tailwind/index.ts:14]:
```suggestion
 * import colors from 'tailwindcss/colors';
 * import starlightPlugin from '@astrojs/starlight-tailwind';
```

Note: this Starlight Tailwind plugin import in this comment also had a typo.

**Review by HiDeoo — COMMENTED** [review]
This looks great, thanks for the update. I guess we also want to switch from `require` to `import` too?

Note: there is also one require in the `docs/src/components/theme-designer.astro` file that I cannot suggest change for it.

**Review by HiDeoo — APPROVED** [review]
Thanks for the changes :raised_hands:

Yeah, I tried it too and was not expecting for it to work ^^ (but I guess it makes sense considering they're using `jiti` under the hood).

Personally, I think this looks great and everything is covered. Let's see if others have any comments.

File: .changeset/thick-bats-pay.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/starlight-tailwind': patch
+---
+
+use `tailwind.config.mjs` file by default config
```
Comment (delucis) [inline .changeset/thick-bats-pay.md:5]:
```suggestion
Show `tailwind.config.mjs` file in docs
```

**Review by delucis — COMMENTED** [review]
Looks great to me! Just suggesting a small tweak to clarify the changeset.

**Review by delucis — APPROVED** [review]
Thanks again @torn4dom4n! LGTM 🚀

**Pull request merged**